src/app/core/registry/dynamic-registry.ts

```diff
@@ -0,0 +1,66 @@
+import { Entity, EntityConstructor } from "../entity/model/entity";
+import { InjectionToken } from "@angular/core";
+import { ComponentType } from "@angular/cdk/overlay";
+import { OnInitDynamicComponent } from "../view/dynamic-components/on-init-dynamic-component.interface";
+
+export class Registry<T> {
+  private map = new Map<string, T>();
+
+  constructor(private beforeAddCheck?: (key: string, mapping: T) => void) {}
+
+  public add(key: string, mapping: T) {
+    this.beforeAddCheck?.(key, mapping);
+    if (this.map.has(key)) {
+      throw Error(
+        `Duplicate entity definition: ${key} is already registered with constructor ${this.map.get(
+          key
+        )}`
+      );
+    }
+    this.map.set(key, mapping);
+  }
+
+  public addAliases(keys: string[], mapping: T) {
+    keys.forEach((key) => {
+      this.add(key, mapping);
+    });
+  }
+
+  public get(key: string): T | undefined {
+    if (!this.has(key)) {
+      console.warn(`Requested item is not registered`);
+    }
+    return this.map.get(key);
+  }
+
+  public has(key: string) {
+    return this.map.has(key);
+  }
+}
+
+export type EntityRegistry = Registry<EntityConstructor>;
+export const ENTITIES = new InjectionToken<EntityRegistry>(
```

Comment:
So there is a trick for not having to use this injection token that I found in the CASL library ([source](https://casl.js.org/v5/en/package/casl-angular#type-script-support)).
It would look something like the following
```suggestion
export const EntityRegistry = Registry as new (beforeAddCheck?: (key: string, mapping: EntityRegistry) => void) => EntityRegistry;
```
Its rather hacky but we can write a readable helper type for it.
The service can be provided like this:
```javascript
{ provide: EntityRegistry, useValue: entityRegistry }
```
and injected like this:
```javascript
constructor(private: entityRegistry: EntityRegistry) {}
```
What do you think @Schottkyc137 ? I can also implement it if you like the approach.

Reply:
If this is working I like the idea. But will this work with multiple registries? If you provide the `EntityRegistry` and a `RouteRegistry` for example, how can angular distinguish between the two?